twine/package.py

```diff
@@ -232,6 +254,11 @@ def from_filename(cls, filename: str, comment: Optional[str]) -> "PackageFile":
         # than 2.4.
         if version.Version(meta.get("metadata_version", "0")) < version.Version("2.4"):
             meta.pop("license_files", None)
+
+        description = meta.get("description", None)
+        if description is not None:
+            meta["description"] = _dedent(description)
```

Comment:
Since this behavior is only necessary for a small and shrinking subset of users, perhaps it should be implemented as one line here (e.g. `_load_description_field(meta)`) and have the docstring describe the (limited) purpose and deadline or condition for when it can be removed.